web-console/knikubevirt/vm-templates/vm-templates.md

```diff
@@ -0,0 +1,86 @@
+- Users can also create templates based on existing VMs from the actions menu of any VM within the VM List View.
+
+
+
+## Template list view
+
+![Offline Template List View](img/1-0-vm-template-list.jpg)
+
+The user will be able to view a list of all templates that exist. From this List, the user can create a new template, create a virtual machine from a selected template, edit a selected template, or delete a selected template.
+
+
+
+
+## Create template from primary button
+
+
+![Offline Template List View](img/1-1-0-create-template.jpg)
+
+Clicking “Create Template” will display a wizard similar to the “No Template state” wizard from Create VM but with slight adjustments.
+
+![Offline Template List View](img/1-1-1-create-template-modal.jpg)
+
+1. Modal title changes to “Create Template”
+2. “Template” is no longer a Provision Source option
+3. The two checkboxes for “Start virtual machine on creation” and “Create new template from configuration” are removed
+4. The quick action button at the bottom is now “Create Template”
+5. In Step 5 “Review”, references to “Virtual Machine” are replaced with “Template”
+
+    a. Create/Creating Virtual Machine -> Create/Creating Template
+
+    b. Virtual Machine creation failed -> Template creation failed
+
+    c. Virtual machine created successfully -> Template created successfully
+
+Everything else remains the same.
+
+
+## Create template from VM List
+
+
+![Offline Template List View](img/1-2-0-create-template-from-vm-list.jpg)
+
+Users can also create templates based on existing VMs from the actions menu of any VM within the VM List View. 
+
+![Offline Template List View](img/1-2-1-create-template-from-vm-list-modal.jpg)
+
+The VM they choose pre-fills the Provision Source, OS, Flavor, Workload Profile, and all other Networking, Storage, and Advanced settings of the Create Template wizard.
+
+## Create VM from another template
+
+
+![Offline Template List View](img/1-3-0-create-template-from-template-list.jpg)
+
+Users can initiate the Create VM wizard from the actions menu of a VM template.
+
+![Offline Template List View](img/1-3-1-create-template-from-template-list-modal.jpg)
+
+The template they choose is preselected and the OS, Flavor, and Workload Profiles are filled in. Refer to the Create VM wizard for all other steps.
+
+
+
+![Offline Template List View](img/1-2-2-create-template-from-vm-list-starting.jpg)
+
+After successfully working through the Create VM Wizard, the user would be taken to the Virtual Machine List View to show the new VM creation progress.
+
+## Template details (coming soon)
+
+Clicking on a template will bring the user to the template’s Overview tab.
+
+The YAML tab allows the user to edit the template’s configuration.
+
+The Virtual Machines tab displays a VM List View with only the VMs that were created using the template. This list behaves mostly the same, but the actions kebab will likely need to be different (no Create Template button, for example)
```

Comment:
should we say "behaves the same with the relevant actions available in the kebab". It is a bit less open ended.

Reply:
Changed that according to your suggestion.
This list relates to a discussion I created in the doc, which I find a bit off:

- I suspect that this list suggests that there is a relation between a template and a child. Because of this suggestion, I misunderstood the doc and thought that it suggests creating a link between the template and the VM created from it (making it so that any changes applied on the template will change the VM). 
- I'm not sure what kind of action will come out of viewing this list.
- Since a template is a mere starting point for creating a VM, some or all VMs created from it will be modified. So, will they be visible here? what're the modifications, threshold?

Reply:
Ya that's a good point. If a VM is created from a template but then is modified would we still want to list that vm as a result of that template? @lizsurette how much do you recall about this design, do you know why we are listing the VMs that were created from the template? It would be good to know the use case.

Reply:
It's probably worth a follow-up with Moran, but we definitely went back and forth between there being a strong relationship between VMs and Templates vs. not doing that. Eventually we decided to start without that strong relationship. Maybe this could be removed? Or maybe made more clear that these VMs were created based on this template, but may have changed a bit since initial creation.